apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/ConnectionStringBuilderTests.cs

```diff
@@ -117,5 +118,51 @@ public void InstanceOnlyDataSource()
             Assert.Equal("instance1", connectionStringBuilder.SpannerInstance);
             Assert.Null(connectionStringBuilder.SpannerDatabase);
         }
+
+        [Fact]
+        public void CredentialFile()
+        {
+            // ReSharper disable once JoinDeclarationAndInitializer
+            string appFolder;
+#if NETCOREAPP1_0
+            appFolder = AppContext.BaseDirectory;
+#endif
+#if !NETCOREAPP1_0
+            appFolder = AppDomain.CurrentDomain.BaseDirectory;
+#endif
+            string jsonFile = $"{appFolder}{Path.DirectorySeparatorChar}SpannerEF-8dfc036f6000.json";
+            Assert.True(File.Exists(jsonFile));
+            using (var connection = new SpannerConnection($"CredentialFile={jsonFile}"))
+            {
+                Assert.NotNull(connection.Credential);
+            }
+        }
+
+        [Fact]
+        public void CredentialFileRelative()
+        {
+            using (var connection = new SpannerConnection("CredentialFile=SpannerEF-8dfc036f6000.json"))
+            {
+                Assert.NotNull(connection.Credential);
+            }
+        }
+
+        [Fact]
+        public void CredentialFileP12Error()
+        {
+            using (var connection = new SpannerConnection("CredentialFile=SpannerEF-8dfc036f6000.p12"))
+            {
+                Assert.Throws<ArgumentException>(() => connection.Credential);
+            }
+        }
+
+        [Fact]
+        public void CredentialFileNotFound()
+        {
+            using (var connection = new SpannerConnection("CredentialFile=..\\BadFilePath.json"))
+            {
+                Assert.Throws<FileNotFoundException>(() => connection.Credential);
```

Comment:
The .NET framework [property design guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/property) suggest not doing this: "AVOID throwing exceptions from property getters.". Can the constructor throw or can this property be turned into a method?

Reply:
good point, I'll switch to a method.